Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -488,66 +585,46 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable, IBufferPr
             throw PythonOps.TypeError("cannot delete memory");
         }
 
-        public object this[[NotNull]Slice slice] {
+        public object? this[[NotNull]Slice slice] {
             get {
                 CheckBuffer();
-                int start, stop, step, count;
-                FixSlice(slice, __len__(), out start, out stop, out step, out count);
-
-                List<int> dimensions = new List<int>();
-
-                // When a multidimensional memoryview is sliced, the slice
-                // applies to only the first dimension. Therefore, other
-                // dimensions are inherited.
-                dimensions.Add(count);
-
-                // In a 1-dimensional memoryview, the difference in bytes
-                // between the position of mv[x] and mv[x + 1] is guaranteed
-                // to be just the size of the data. For multidimensional
-                // memoryviews, we must worry about the width of all the other
-                // dimensions for the difference between mv[(x, y, z...)] and
-                // mv[(x + 1, y, z...)]
-                int firstIndexWidth = _itemsize;
-                for (int i = 1; i < shape.__len__(); i++) {
-                    int dimensionWidth = Converter.ConvertToInt32(shape[i]);
-                    dimensions.Add(dimensionWidth);
-                    firstIndexWidth *= dimensionWidth;
+                if (_numDims == 0) {
+                    throw PythonOps.TypeError("invalid indexing of 0-dim memory");
                 }
 
-                int newStart = _start + start * firstIndexWidth;
-                int newEnd = _start + stop * firstIndexWidth;
-                int newStep = ((long)_step * step).ClampToInt32();
```

Comment:
I guess we can get rid of `PythonOps.ClampToInt32` now? 😏

Reply:
I almost used it in 6a6841d2...
It can go now, but it may come back depending on the approach to #52.